rosbag2_cpp/src/rosbag2_cpp/writers/sequential_writer.cpp

```diff
@@ -204,6 +259,7 @@ void SequentialWriter::write(std::shared_ptr<rosbag2_storage::SerializedBagMessa
   const auto duration = message_timestamp - metadata_.starting_time;
   metadata_.duration = std::max(metadata_.duration, duration);
 
+  compress_message(message);
```

Comment:
Do you really gain anything by compressing twice, once at the message level and once at the file level? There must be something in information theory that says this is not very effective.

Reply:
The compression option passed to the reader only allows you to compress either by file or message. It can't be both.

Reply:
Oh, never mind. The `CompressionMode` can only be `FILE` or `MESSAGE`.

Reply:
It'd be more clear if you did the compression mode check outside of `compress_message()` and `compress_file_and_update_metadata()`. You can leave the `compressor_ != nullptr` check inside those functions though.